src/iocomponent.rs

```diff
@@ -0,0 +1,312 @@
+use std::ffi::OsStr;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use image::{GenericImage, GenericImageView, Rgba, RgbaImage};
+use serde::{Deserialize, Serialize};
+use tokio::runtime::Runtime;
+
+use crate::backend::Backend;
+use crate::chunk::Chunk;
+use crate::paint_canvas::PaintCanvas;
+use crate::xcoder::Xcoder;
+use crate::Error;
+
+/// The format version in a `.netcanv`'s `canvas.toml` file.
+pub const CANVAS_TOML_VERSION: u32 = 1;
+
+/// A `canvas.toml` file.
+#[derive(Serialize, Deserialize)]
+struct CanvasToml {
+   /// The format version of the canvas.
+   version: u32,
+}
+
+pub struct IoComponent {
+   runtime: Arc<Runtime>,
+
+   /// The path to the `.netcanv` directory this paint canvas was saved to.
+   filename: Option<PathBuf>,
+}
+
+impl IoComponent {
+   pub fn new(runtime: Arc<Runtime>) -> Self {
+      IoComponent {
+         runtime,
+         filename: None,
+      }
+   }
+
+   /// Saves the entire paint to a PNG file.
```

Comment:
I think there's a spelling mistake here:
```suggestion
   /// Saves the entire paint canvas to a PNG file.
```